phpBB/phpbb/template/twig/lexer.php

```diff
@@ -118,7 +118,7 @@ public function tokenize($code, $filename = null)
 
 		// Replace all of our variables, {VARNAME}, with Twig style, {{ VARNAME }}
 		// Appends any filters
-		$code = preg_replace('#{([a-zA-Z0-9_\.]+)(\|[^}]+?)?}#', '{{ $1$2 }}', $code);
+		$code = preg_replace('#{([a-zA-Z_][a-zA-Z0-9_\.]*)(\|[^}]+?)?}#', '{{ $1$2 }}', $code);
```

Comment:
I see you want to make varnames starting only with the alpha letters or underscore, but this new regex will match ```test``` in the string ```123test``` for example.
I think you'd need to add the caret anchor to indicate the starting string letter, like ```(^[a-zA-Z_][a-zA-Z0-9_\.]*)```. Thus, it will need adding more specific tests I'd say.